monitoring/src/main/java/io/silverware/microservices/monitoring/MonitoringMethodHandler.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * -----------------------------------------------------------------------\
+ * SilverWare
+ *
+ * Copyright (C) 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -----------------------------------------------------------------------/
+ */
+package io.silverware.microservices.monitoring;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import javax.annotation.Priority;
+import javax.enterprise.inject.spi.InjectionPoint;
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+
+import java.lang.management.ManagementFactory;
+import java.lang.reflect.Method;
+import java.math.BigDecimal;
+
+import io.silverware.microservices.monitoring.annotations.NotMonitored;
+import io.silverware.microservices.providers.cdi.internal.MicroserviceMethodHandler;
+import io.silverware.microservices.providers.cdi.internal.MicroserviceProxyBean;
+
+/**
+ * This is a method handler for monitoring. It extends {@link MicroserviceMethodHandler} and its priority is {@link MicroserviceMethodHandler#DEFAULT_PRIORITY} - 1.
+ */
+@Priority(MicroserviceMethodHandler.DEFAULT_PRIORITY - 1)
```

Comment:
If I understand it correctly, this method handler should be (and is) executed after `HystrixMethodHandler` in order not to count Hystrix overhead into execution time, right? In that case I would put here priority `50` so there is enough space to define some more method handlers between `HystrixMethodHandler` (`0`) and `MonitoringMethodHandler` as well as between `MonitoringMethodHandler` and handlers with default priority (`100`).